x-pack/platform/plugins/shared/lens/public/react_embeddable/initializers/initialize_dashboard_services.ts

```diff
@@ -79,7 +79,7 @@ export function initializeDashboardServices(
   // For some legacy reason the title and description default value is picked differently
   // ( based on existing FTR tests ).
   const defaultTitle$ = new BehaviorSubject<string | undefined>(
-    initialState.title || internalApi.attributes$.getValue().title
+    internalApi.attributes$.getValue().title
```

Comment:
Maybe I'm missing something, but I think we just need to change the `defaultTitle` to be based on the title provided by attributes in the initialState. This would be the title as returned from the Lens saved object. And `initialState.attributes.title` should not be mutable.

```suggestion
    initialState.attributes.title || internalApi.attributes$.getValue().title
```